### DISCUSSION TITLE
Get meta data inside a regular component

### DISCUSSION THREAD

**Discussion by K-Mazur** [vercel/next.js#53674] [Help]
### Summary

Hi,

I'm generating the meta data of all pages with the generateMetadata function. Inside of it I'm making an async request to the backend. Is there some hook or a helper function that can read the meta data inside components?

Let's say that I have a server side component that renders an <h1> with page title. Do I need to to make an additional request to the server to get the title or can I somehow read the meta data values?

**Comment by icyJoseph** [comment]
I think it should be ok to fetch again, as long as it is all inside server components, the `fetch` function will cache/dedupe similar requests, and give you the same answer.

You could also I guess, set a global object, and read it where you need to, during SSR:

- https://nextjs.org/docs/getting-started/react-essentials#sharing-data-between-server-components

> [Sharing data between Server Components](https://nextjs.org/docs/getting-started/react-essentials#sharing-data-between-server-components)
Since Server Components are not interactive and therefore do not read from React state, you don't need React context to share data. Instead, you can use native JavaScript patterns for common data that multiple Server Components need to access. For example, a module can be used to share a database connection across multiple components:

Be careful with this though: https://github.com/vercel/next.js/discussions/52934#discussioncomment-6498597
